pandas/_libs/lib.pyx

```diff
@@ -145,6 +145,32 @@ i8max = <int64_t>INT64_MAX
 u8max = <uint64_t>UINT64_MAX
 
 
+cdef bint PYARROW_INSTALLED = False
+
+try:
+    import pyarrow as pa
+
+    PYARROW_INSTALLED = True
+except ImportError:
+    pa = None
+
+
+cpdef is_pyarrow_array(obj):
+    if PYARROW_INSTALLED:
+        return isinstance(obj, (pa.Array, pa.ChunkedArray))
+    return False
+
+
+cpdef is_pyarrow_scalar(obj):
+    if PYARROW_INSTALLED:
+        return isinstance(obj, pa.Scalar)
```

Comment:
for my edification, if we did have pyarrow as a required dependency is there a more performant version of this check?  IIRC you said something about lack of ABI stability so that might not be viable?

Reply:
No, this is exactly the same as what we would do if pyarrow is a required dependency, except for the `if PYARROW_INSTALLED:` check (but that should be fast, since that's a boolean).

Only if we would add a compile-time dependency on pyarrow, we could `cimport` the class, and that might be faster then? (but that's not included in the current discussion, and this specific case for checking scalars is certainly in itself not worth a compile time dependency)

Reply:
> Only if we would add a compile-time dependency on pyarrow

Yah that's the one i was wondering about.